app/main/views/suppliers.py

```diff
@@ -471,3 +481,80 @@ def update_supplier_framework_details(supplier_id, framework_slug):
+    # (technically we should be doing a FOR UPDATE of SupplierFramework and a FOR SHARE of Framework)
+    interest_record = SupplierFramework.query.filter(
+        SupplierFramework.supplier_id == supplier.supplier_id,
+        SupplierFramework.framework_id == framework.id
+    ).first()
+    if not interest_record:
+        abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
+
+    if not interest_record.on_framework:
+        abort(404, "supplier_id '{}' is not on framework {}".format(supplier_id, framework_slug))
+
+    # acceptable variation_slugs must have corresponding createdAt entry in variations dict
+    if not (framework.framework_agreement_details or {}).get("variations", {}).get(variation_slug, {}).get("createdAt"):
+        abort(404, "Unknown variation '{}' for framework {}".format(variation_slug, framework_slug))
+
+    if (interest_record.agreed_variations or {}).get(variation_slug, {}).get("agreedAt"):
+        abort(400, "supplier_id '{}' has already agreed to variation '{}'.".format(supplier_id, variation_slug))
+
+    json_payload = get_json_from_request()
+    updater_json = validate_and_return_updater_request()
+    json_has_required_keys(json_payload, ["agreedVariation"])
+    json_only_has_required_keys(json_payload["agreedVariation"], ["agreedUserId"])
+
+    user = User.query.filter(User.id == json_payload["agreedVariation"]["agreedUserId"]).first()
+    if not user:
+        abort(400, "No user found with id '{}'".format(json_payload["agreedVariation"]["agreedUserId"]))
+    if supplier_id != user.supplier_id:
+        abort(403, "user '{}' isn't associated with supplier_id '{}'".format(
+            json_payload["agreedVariation"]["agreedUserId"],
+            supplier_id,
+        ))
+
+    agreed_variations = interest_record.agreed_variations.copy() if interest_record.agreed_variations else {}
+    agreed_variations[variation_slug] = agreed_variations.get(variation_slug, {})
+    agreed_variations[variation_slug].update({
+        "agreedAt": datetime.utcnow().strftime(DATETIME_FORMAT),
+        "agreedUserId": user.id,
+    })
+    interest_record.agreed_variations = agreed_variations
+
+    audit_event = AuditEvent(
+        audit_type=AuditTypes.agree_framework_variation,
+        user=updater_json['updated_by'],
+        data={
+            'supplierId': supplier.supplier_id,
+            'frameworkSlug': framework_slug,
+            'variationSlug': variation_slug,
+            'update': json_payload["agreedVariation"],
+        },
+        db_object=interest_record,
+    )
+
+    try:
+        db.session.add(interest_record)
+        db.session.add(audit_event)
+        db.session.commit()
+    except IntegrityError as e:
+        db.session.rollback()
+        return jsonify(message="Database Error: {0}".format(e)), 400
+
+    return jsonify(
+        agreedVariation=SupplierFramework.serialize_agreed_variation(agreed_variations[variation_slug]),
```

Comment:
Also, thinking about this a bit more (insomnia's a bitch), I think the route should return the entire `SupplierFramework` object, not just the `agreedVariations` part, as that is the model object that's referenced by the URL and that is updated by the call, i.e. `return jsonify(frameworkInterest=interest_record.serialize())`

Other routes that update this record do that (except PUTting a declaration which only returns the declaration, but I think that's probably doing the wrong thing).